### ISSUE TITLE
BasiliskII macOS uses /opt/homebrew/*/libgmp.10.dylib

### ISSUE THREAD

**Issue by RonaldPR** [kanjitalk755/macemu#94]
BasiliskII now does build fine and runs fine on my M1 Mac.

But after notarizing the way I did so far, I cannot launch the notarized BasiliskII:

```
Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x5] Code Signature

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: /opt/homebrew/*/libgmp.10.dylib
  Referenced from: /Users/USER/*/BasiliskII.app/Contents/MacOS/BasiliskII
  Reason: no suitable image found.  Did find:
	/opt/homebrew/*/libgmp.10.dylib: code signature in (/opt/homebrew/*/libgmp.10.dylib) not valid for use in process using Library Validation: mapped file has no Team ID and is not a platform binary (signed with custom identity or adhoc?)
	/opt/homebrew/*/libgmp.10.dylib: stat() failed with errno=1

Binary Images:
       0x1042c8000 -        0x104363fff +com.emaculation.basiliskii (Basilisk II 1.0, SDL2 port - 0) <CC3DCDCC-5B29-303E-8914-F9D2C81E43C9> /Users/USER/*/BasiliskII.app/Contents/MacOS/BasiliskII
       0x104534000 -        0x1045b3fff  dyld (852) <CF624584-51D6-329D-91C5-6BEDFADFB2D2> /usr/lib/dyld
```
If I notarize while disabling library validation in order for the app to be able to use third-party frameworks outside the bundle, the resulting BasiliskII app runs fine again.

**Would this mean that BasiliskII will not run on a machine that does not have homebrew installed?
If that is so, this build is not suitable for distribution.**

**Comment by kanjitalk755** [comment]
Sure, building with homebrew requires users to install the libraries as well.
Therefore, I changed to the method of manually installing the static libraries.

**Comment by RonaldPR** [comment]
Build now fails with 999+ warnings and 79 errors.

**Comment by RonaldPR** [comment]
OK, I missed the new instructions.

I do not understand the note:
For Intel Mac, checkout has_fpu_bug branch. But it has FPU issue if the binary runs on Apple Silicon Mac.

What does that mean if I build a universal build on Apple Silicon? Will that build run with correct FPU on both Intel and Apple Silicon?

**Comment by RonaldPR** [comment]
gmp-6.2.1 make check error:

```
libtool:   error: cannot find the library 'libraries/gmp-6.2.1/libgmp.la' or unhandled argument 'libraries/gmp-6.2.1/libgmp.la'
make[4]: *** [t-bswap] Error 1
make[3]: *** [check-am] Error 2
make[2]: *** [check-recursive] Error 1
make[1]: *** [check-recursive] Error 1
make: *** [check] Error 2
```

**Comment by RonaldPR** [comment]
gmp-6.2.1 succeeded without errors the second time I tried.

But now I have a configure error for mpfr-4.1.0:
configure: error: libgmp not found or uses a different ABI (including static vs shared).

(Both gmp-6.2.1 and mpfr-4.1.0 folders are in the Downloads folder.)

**Comment by kanjitalk755** [comment]
> What does that mean if I build a universal build on Apple Silicon? Will that build run with correct FPU on both Intel and Apple Silicon?

If you build the master branch on an Apple Silicon machine, it works without the FPU issue on both Intel and Apple Silicon.

> configure: error: libgmp not found or uses a different ABI (including static vs shared).

You should install GMP before making MPFR.

**Comment by RonaldPR** [comment]
I did the commands as descibed in the instructions, including finally `sudo make install`
How do I install GMP?

**Comment by RonaldPR** [comment]
I ran `sudo make install` for gmp again. In Terminal I could see that gmp was installed in usr/local/

I than ran `./configure --disable-shared` for mpfr again. Same error: libgmp not found

**Comment by RonaldPR** [comment]
gmp is installed in the same place where it was earlier installed with the brew command: /usr/local/Cellar/

There is also a folder /usr/local/opt/ with in it aliases gmp and mpfr, pointing to /usr/local/Cellar/gmp/6.2.1/ and /usr/local/Cellar/mpfr/4.1.0/ respectively.

**Comment by RonaldPR** [comment]
And of course gmp and mpfr are also still available in /opt/homebrew/Cellar/

**Comment by kanjitalk755** [comment]
> configure: error: libgmp not found or uses a different ABI (including static vs shared).

Probably the error is not "not found" but "uses a different ABI".
You should delete all GMP and MPFR libraries installed by Homebrew.

**Comment by RonaldPR** [comment]
Well, that appeared not so simple. I deleted  GMP and MPFR from both usr/local/ and /opt/homebrew. But that left countless broken aliases in subfolders in both locations. Especially installing MPFR failed each time when a file needed to be installed in a place where a broken alias with the same name already existed (with the not so informative error "file or folder does not exist"). I finally managed to remove all aliases that blocked installation, but there are still more left in different places. I hope they will not interfere with other installations or other software.

(I wonder if the issue would not have existed if the project could have followed aliases. For all the GMP and MPFR files that are now installed, aliases existed that pointed to the actual files that were already installed.)

Building and notarizing BasiliskII now worked and the resulting app runs fine here.
